src/transformers/utils/quantization_config.py

```diff
@@ -1232,6 +1233,44 @@ def get_loading_attributes(self):
         return loading_attibutes_dict
 
 
+@dataclass
+class HiggsConfig(QuantizationConfigMixin):
+    """
+    HiggsConfig is a configuration class for quantization using the HIGGS method.
+
+    Args:
+        bits (int, *optional*, defaults to 4):
+            Number of bits to use for quantization. Default is 4.
+        p (int, *optional*, defaults to 2):
+            Parameter for the HIGGS quantization method. Default is 2.
+        linear_weights_not_to_quantize (`list`, *optional*, default to ["lm_head.weight"]):
+            List of linear weight names that should not be quantized.
+    """
+
+    def __init__(
+        self,
+        bits: int = 4,
+        p: int = 2,
+        linear_weights_not_to_quantize: Optional[List[str]] = None,
+        **kwargs,
+    ):
+        if linear_weights_not_to_quantize is None:
+            linear_weights_not_to_quantize = ["lm_head.weight"]
+        self.quant_method = QuantizationMethod.HIGGS
+        self.bits = bits
+        self.p = p
+        self.linear_weights_not_to_quantize = linear_weights_not_to_quantize
+        self.num_sms_packed = 128
```

Comment:
Can you add a description of what this is used for ? The user shouldn't have to worry about that ?

Reply:
Updated the docstring to better reflect what those are